File: AviotSystemv0.0.1/app/src/main/java/com/gesieniec/przemyslaw/aviotsystemv001/iothandler/devices/CommonDevice.java
package com.gesieniec.przemyslaw.aviotsystemv001.iothandler.devices;

import android.util.Log;

import com.gesieniec.przemyslaw.aviotsystemv001.iothandler.DeviceAction;
import com.gesieniec.przemyslaw.aviotsystemv001.iothandler.DeviceCapabilities;
import com.gesieniec.przemyslaw.aviotsystemv001.iothandler.DeviceType;

import java.net.InetAddress;
import java.util.HashMap;
import java.util.List;

/**
 * Created by przem on 08.11.2017.
 */

public abstract class CommonDevice {


    /**
     * fields
     */
    protected String name;
    protected String location;
    protected String namePL;
    protected String locationPL;
    protected InetAddress deviceAddress;
    protected String macAddress;
    protected HashMap<String,DeviceAction> actionMapENG;
    protected HashMap<String,DeviceAction> actionMapPL;
    protected List<String> actionList;
    protected boolean isDataUpdated = false;
    /**
     * getters
     */
    public String getName() {
        return name;
    }
    public String getLocation() {
        return location;
    }
    public InetAddress getDeviceAddress() {
        return deviceAddress;
    }
    public String getMacAddress() {
        return macAddress;
    }
    public String getNamePL() {
        return namePL;
    }
    public String getLocationPL() {
        return locationPL;
    }
    public List<String> getActionList() {
        return actionList;
    }

    /**
     * setters
     */
    public void setNamePL(String namePL) {
        this.namePL = namePL;
    }
    public void setLocationPL(String locationPL) {
        this.locationPL = locationPL;
    }
    public void setName(String name) {
        this.name = name;
    }
    public void setLocation(String location) {
        this.location = location;
    }


    public CommonDevice(String name, String location,InetAddress deviceAddress,String macAddress) {
        this.name = name;
        this.location = location;
        this.deviceAddress = deviceAddress;
        this.macAddress = macAddress;
        Log.d("CommonDevice: ","New device:" );
    }
    public abstract DeviceType getDeviceType();
    public abstract HashMap<String, DeviceAction> getActionMapENG();
    public abstract HashMap<String, DeviceAction> getActionMapPL();
//    public abstract void updateCommonDataClass();
    public abstract String getMessageToSend(DeviceCapabilities capabilities);
    public abstract void updateDeviceWithCapabilities(DeviceCapabilities deviceCapabilities);
}

File: ArduinoCode/esp12E/CommonDevice.cpp
#include "CommonDevice.h"

CommonDevice::CommonDevice() {}
CommonDevice::CommonDevice(String name, String location, String macAddress)
{
	this->name = name;
	this->location = location;
	this->macAddress = macAddress;
}


CommonDevice::~CommonDevice()
{
}

//
//GETTERS
//
String CommonDevice::getMessageType()
{
  return messageType;
}

String CommonDevice::getName()
{
	return name;
}

String CommonDevice::getLocation()
{
	return location;
}
String CommonDevice::getMsg()
{
  return msg;
}
String CommonDevice::getMacAddress()
{
	return macAddress;
}

//
//SETTERS
//
void CommonDevice::setName(String name)
{
	this->name = name;
}
void CommonDevice::setMsg(String msg)
{
  this->msg = msg;
}
void CommonDevice::setLocation(String location)
{
	this->location = location;
}

void CommonDevice::setMacAddress(String macAddress)
{
	this->macAddress = macAddress;
}
void CommonDevice::setMessageType(String messageType)
{
  this->messageType = messageType;
}

String CommonDevice::macToStr(const uint8_t* mac)
{
  String result;
  for (int i = 0; i < 6; ++i) {
    result += String(mac[i], 16);
    if (i < 5)
      result += ':';
  }
  return result;
}
String CommonDevice::boolToString(bool value)
{
  return value ? "true" : "false";
}

